code/game/machinery/dna_infuser/organ_sets/fish_organs.dm

```diff
@@ -0,0 +1,324 @@
+	owner.AddElementTrait(TRAIT_WADDLING, type, /datum/element/waddling)
+	RegisterSignal(owner, COMSIG_MOVABLE_MOVED, PROC_REF(check_location))
+	check_location(owner, owner.loc)
+
+/obj/item/organ/external/tail/fish/on_mob_remove(mob/living/carbon/owner)
+	. = ..()
+	owner.remove_traits(list(TRAIT_WADDLING, TRAIT_NO_STAGGER), type)
+	owner.remove_movespeed_modifier(/datum/movespeed_modifier/fish_on_water)
+	owner.remove_actionspeed_modifier(/datum/actionspeed_modifier/fish_on_water)
+	UnregisterSignal(owner, COMSIG_MOVABLE_MOVED)
+
+/obj/item/organ/external/tail/fish/proc/check_location(mob/living/carbon/source, atom/movable/old_loc, dir, forced)
+	var/was_water = istype(old_loc, /turf/open/water)
+	var/is_water = istype(source.loc, /turf/open/water) && !HAS_TRAIT(source.loc, TRAIT_TURF_IGNORE_SLOWDOWN)
+	if(was_water && !is_water)
+		source.remove_movespeed_modifier(/datum/movespeed_modifier/fish_on_water)
+		source.remove_actionspeed_modifier(/datum/actionspeed_modifier/fish_on_water)
+		source.add_traits(list(TRAIT_OFF_BALANCE_TACKLER, TRAIT_NO_STAGGER, TRAIT_NO_THROW_HITPUSH), type)
+	else if(!was_water && is_water)
+		source.add_movespeed_modifier(/datum/movespeed_modifier/fish_on_water)
+		source.add_actionspeed_modifier(/datum/actionspeed_modifier/fish_on_water)
+		source.add_traits(list(TRAIT_OFF_BALANCE_TACKLER, TRAIT_NO_STAGGER, TRAIT_NO_THROW_HITPUSH), type)
+
+/datum/bodypart_overlay/mutant/tail/fish
+	feature_key = "fish_tail"
+	color_source = ORGAN_COLOR_HAIR
+
+/datum/bodypart_overlay/mutant/tail/fish/on_mob_insert(obj/item/organ/parent, mob/living/carbon/receiver)
+	//Initialize the related dna feature block if we don't have any so it doesn't error out.
+	//This isn't tied to any species, but I kinda want it to be mutable instead of having a fixed sprite accessory.
+	if(imprint_on_next_insertion && !receiver.dna.features["fish_tail"])
+		receiver.dna.features["fish_tail"] = pick(SSaccessories.tails_list_fish)
+		receiver.dna.update_uf_block(DNA_FISH_TAIL_BLOCK)
+	return ..()
+
+/datum/bodypart_overlay/mutant/tail/fish/get_global_feature_list()
+	return SSaccessories.tails_list_fish
+
+
+///Lungs that replace the need of oxygen with water vapor or being wet
+/obj/item/organ/internal/lungs/fish
+	name = "mutated gills"
+	desc = "Fish DNA infused on what once was a normal pair of lungs that now require spacemen to breathe water vapor, or keep themselves covered in water."
+	icon = 'icons/obj/medical/organs/infuser_organs.dmi'
+	icon_state = "gills"
+
+	safe_oxygen_min = 0 //We don't breathe this
+	///The required partial pressure of water_vapor for not drowing
+	var/safe_water_level = 29
+	///The special bubble icon that we give to mobs with this organ. It's a indie rpg reference btw
+	var/datum/component/bubble_icon_override/bubble_icon
+
+/obj/item/organ/internal/lungs/fish/Initialize(mapload)
+	//This takes precedence over oygen for the amphibious subtype
+	add_gas_reaction(/datum/gas/water_vapor, always = PROC_REF(breathe_water))
```

Comment:
there's no reason to do this early no parent instance exists. if you are concerned about duplicates consider making the proc error (or stack trace?) if one already exists

Reply:
no reason to do it before the parent call essentially

Reply:
Ok, imma rearrange it a bit and add some clarity to the comment, but basically, for the amphibious subtype of this organ,  we want the breathe_water proc to run before breathe_oxygen, so that if breathe waters succeeds, breathe oxygen will just early return.